src/functional/iterator/transforms/pass_manager.py

```diff
@@ -1,27 +1,49 @@
+import enum
+
 from functional.iterator import ir
 from functional.iterator.transforms.cse import CommonSubexpressionElimination
 from functional.iterator.transforms.global_tmps import CreateGlobalTmps
 from functional.iterator.transforms.inline_fundefs import InlineFundefs, PruneUnreferencedFundefs
 from functional.iterator.transforms.inline_lambdas import InlineLambdas
 from functional.iterator.transforms.inline_lifts import InlineLifts
 from functional.iterator.transforms.normalize_shifts import NormalizeShifts
+from functional.iterator.transforms.simple_inline_heuristic import heuristic
 from functional.iterator.transforms.unroll_reduce import UnrollReduce
 
 
+@enum.unique
+class LiftMode(enum.Enum):
+    FORCE_INLINE = enum.auto()
+    FORCE_TEMPORARIES = enum.auto()
+    SIMPLE_HEURISTIC = enum.auto()
+
+
+def _inline_lifts(ir, lift_mode):
+    if lift_mode == LiftMode.FORCE_INLINE:
+        return InlineLifts().visit(ir)
+    if lift_mode == LiftMode.SIMPLE_HEURISTIC:
+        predicate = heuristic(ir)
+        return InlineLifts(predicate).visit(ir)
+    assert lift_mode == LiftMode.FORCE_TEMPORARIES
+    return ir
+
+
 def apply_common_transforms(
     ir: ir.Node,
-    use_tmps=False,
+    lift_mode=None,
```

Comment:
```suggestion
    **,
    lift_mode=None,
```
We should not pass all these options as positional arguments. This might break something, therefore we should do it in a separate PR.

Reply:
Didn’t break anything, so I forced keyword-only args.